propertyestimator/protocols/simulation.py

```diff
@@ -332,57 +346,477 @@ class BaseYankProtocol(BaseProtocol):
+
+        if charge_method.lower() != 'pme':
+            raise ValueError('Currently only PME electrostatics are supported.')
+
+        # Do we need to include `ewald_error_tolerance`?
+        return {'default': {
+            'nonbonded_method': charge_method,
+            'nonbonded_cutoff': quantity_to_string(cutoff.in_units_of(unit.nanometers)),
+
+            'switch_distance': quantity_to_string(switch_distance.in_units_of(unit.nanometers)),
+        }}
+
+    def _get_system_dictionary(self):
+        """Returns a dictionary of the system which will be serialized
+        to a yaml file and passed to YANK. Only a single system may be
+        specified.
+
+        Returns
+        -------
+        dict of str and Any
+            A yaml compatible dictionary of YANK systems.
+        """
+        raise NotImplementedError()
+
+    def _get_protocol_dictionary(self):
+        """Returns a dictionary of the protocol which will be serialized
+        to a yaml file and passed to YANK. Only a single protocol may be
+        specified.
+
+        Returns
+        -------
+        dict of str and Any
+            A yaml compatible dictionary of a YANK protocol.
+        """
+        raise NotImplementedError()
+
+    def _get_full_input_dictionary(self, available_resources):
+        """Returns a dictionary of the full YANK inputs which will be serialized
+        to a yaml file and passed to YANK
+
+        Parameters
+        ----------
+        available_resources: ComputeResources
+            The resources available to execute on.
+
+        Returns
+        -------
+        dict of str and Any
+            A yaml compatible dictionary of a YANK input file.
+        """
+
+        system_dictionary = self._get_system_dictionary()
+        system_key = next(iter(system_dictionary))
+
+        protocol_dictionary = self._get_protocol_dictionary()
+        protocol_key = next(iter(protocol_dictionary))
+
+        return {
+            'options': self._get_options_dictionary(available_resources),
+
+            'solvents': self._get_solvent_dictionary(),
+
+            'systems': system_dictionary,
+            'protocols': protocol_dictionary,
+
+            'experiments': {
+                'system': system_key,
+                'protocol': protocol_key
+            }
+        }
+
+    @staticmethod
+    def _extract_trajectory(checkpoint_path, output_trajectory_path):
+        """Extracts the stored trajectory of the 'initial' state from a
+        yank `.nc` checkpoint file and stores it to disk as a `.dcd` file.
+
+        Parameters
+        ----------
+        checkpoint_path: str
+            The path to the yank `.nc` file
+        output_trajectory_path: str
+            The path to store the extracted trajectory at.
+        """
+
+        from yank.analyze import extract_trajectory
+
+        mdtraj_trajectory = extract_trajectory(checkpoint_path, state_index=0)
```

Comment:
A couple of suggestions:
- If this is a periodic simulation (I think it is), I'd set `image_molecules=True`.
- If you're not interested in water, I'd set `keep_solvent=False`

See also API docs: http://getyank.org/latest/api/analysis_api.html#yank.analyze.extract_trajectory

Reply:
Thanks for the suggestions! I think we need the water for reweighting, but `image_molecules` definitely looks useful!